tests/Data/BlockChainData/Test_BlockChain.cpp

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (C) 2019 Zilliqa
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include <array>
+#include <string>
+#include <vector>
+
+#include "libCrypto/Sha2.h"
+#include "libData/BlockChainData/BlockChain.h"
+#include "libData/BlockChainData/BlockLinkChain.h"
+#include "libTestUtils/TestUtils.h"
+#include "libUtils/DataConversion.h"
+#include "libUtils/Logger.h"
+
+#define BOOST_TEST_MODULE blockchaintest
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_ALTERNATIVE_INIT_API
+
+#include <boost/test/unit_test.hpp>
+
+using namespace std;
+using namespace boost::multiprecision;
+
+BOOST_AUTO_TEST_SUITE(blockchaintest)
+
+// Fills vector of BlockLink starting at position _from_ with random number of
+// BlockLink elements in range <_min_,_max_ >. Maximum is size(uint8_t).
+void appendBlockLinkAndChain_v(BlockLinkChain& blc, vector<BlockLink>& bl_v,
+                               const uint8_t& min, const uint8_t& max) {
+  if (max < min) {
+    throw "Invalid range, to < from + 1";
+  }
+
+  uint8_t lastIndex = bl_v.size();
+
+  for (uint8_t i = lastIndex;
+       i < lastIndex + TestUtils::RandomIntInRng<uint8_t>(min, max); i++) {
```

Comment:
Will `RandomIntInRng` be re-evaluated for every iteration?  Might be better to move out the expression `lastIndex + RandomIntInRng` as a constant or variable?

Reply:
Ou, of course it will, thanks and fixed!